ts/private/ts_validate_options.bzl

```diff
@@ -23,6 +24,20 @@ def _validate_options_impl(ctx):
     # We make it a .d.ts file so we can plumb it to the deps of the ts_project compile.
     marker = ctx.actions.declare_file("%s.optionsvalid.d.ts" % ctx.label.name)
 
+    # Provider validation
+    if not ctx.attr.allow_js:
```

Comment:
I think checkJs is the feature we ought to make this conditional on, see the definition of the two of them.

However that would mean adding an attribute to ts_project, and maybe allowJs is always true when checkJs is true (you could verify that's what `tsc` does) so then allow_js here just makes our disabling a bit broader and it's fine

Reply:
any thought on this one before we merge?